packages/material-ui-lab/src/Slider/Slider.js

```diff
@@ -0,0 +1,515 @@
+      },
+    },
+    /* Thumb styles  */
+    thumb: {
+      position: 'absolute',
+      zIndex: 2,
+      transform: 'translate(-50%, -50%)',
+      width: 12,
+      height: 12,
+      borderRadius: '50%',
+      transition: commonTransitions,
+      backgroundColor: colors.primary,
+      '&:focus': {
+        boxShadow: `0px 0px 0px 10px ${fade(colors.primary, 0.3)}`,
+      },
+      '&$active': {
+        width: 18,
+        height: 18,
+        '&:focus': {
+          boxShadow: 'none',
+        },
+        '&$initial': {
+          border: `2px solid ${colors.focused}`,
+        },
+      },
+      '&$disabled': {
+        width: 10,
+        height: 10,
+        cursor: 'no-drop',
+        backgroundColor: colors.disabled,
+        '&:focus': {
+          border: `2px solid ${colors.disabled}`,
+          boxShadow: 'none',
+        },
+      },
+      '&$initial': {
+        border: `2px solid ${colors.disabled}`,
+        backgroundColor: 'transparent',
+        '&:focus': {
+          border: `2px solid ${colors.focused}`,
+          boxShadow: `0px 0px 0px 10px ${fade(colors.disabled, 0.3)}`,
+        },
+      },
+    },
+    initial: {},
+    vertical: {},
+    disabled: {},
+    focus: {},
+    active: {},
+    reverse: {},
+  };
+};
+
+const KEY_CODES = {
+  ARROW_LEFT: 37,
+  ARROW_UP: 38,
+  ARROW_RIGHT: 39,
+  ARROW_DOWN: 40,
+  HOME: 36,
+  END: 35,
+};
+
+function normalizeValue(value, min = 0, max = 100) {
+  return Math.min(Math.max(value, min), max);
+}
+
+const percentToValue = ({ percent, max, min }) => {
+  return (max - min) * percent / 100 + min;
+};
+
+const roundToStep = ({ currentValue, newValue, step }) => {
+  const diff = newValue - currentValue;
+
+  if (Math.abs(diff) < step / 2) {
+    return currentValue;
+  }
+
+  const remainder = diff > 0 && newValue % step > 0 ? step - newValue % step : -newValue % step;
+
+  return newValue + remainder;
+};
+
+function getRelativeOffset(event) {
+  const { scrollY, scrollX } = global;
+  const { changedTouches, currentTarget } = event;
+  const { pageX, pageY } = (changedTouches && changedTouches[0]) || event;
+  const { left, top } = currentTarget.getBoundingClientRect();
+
+  return {
+    top: pageY - top - scrollY,
+    left: pageX - left - scrollX,
+  };
+}
+
+const calculatePercent = (event, vertical) => {
+  const { currentTarget } = event;
+  const { width, height } = currentTarget.getBoundingClientRect();
+
+  const offset = getRelativeOffset(event);
+
+  const position = vertical ? offset.top : offset.left;
+  const size = vertical ? height : width;
+
+  return normalizeValue(position / (size / 100), 0, 100);
+};
+
+const timeout = delay => () => new Promise(resolve => setTimeout(resolve, delay));
+
+class Slider extends React.Component {
+  state = { isMove: false, isActive: false, isFocus: false };
+
+  componentDidMount() {
+    global.addEventListener('touchstart', this.touchStart, { passive: false });
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
Could you update this for React 16.3? We're using the polyfill for the new lifecycle methods to support earlier versions.